### DISCUSSION TITLE
[Feature] Add Album Filter to Search Tab

### DISCUSSION THREAD

**Discussion by murtaza-nasir** [immich-app/immich#12258] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

## Description
Add an album filter to the search tab that can be combined with the existing person filter.

## Use Case
To efficiently find photos of specific people within particular albums.

## Proposed Solution
Implement a album filter (dropdown or autocomplete) in the search tab that works alongside other filters, especially the person filter.

## Benefits
- Faster, more targeted searches
- Easily find people in specific contexts/events
- Improved management of large photo libraries

## Potential Challenge
Ensuring efficient performance when combining multiple filters. 

Thank you for considering this feature request.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Immich doesn't have something called a "gallery". Do you mean filtering for albums?

**Reply by murtaza-nasir** [reply]
Yes, I'm sorry I meant albums. I've edited the post for clarity.

**Comment by marcothms** [comment]
I would also love to see this, as I create albums based on events.
Being able to filter my search to just one album would be such a nice addition!

Immich already allows to filter for photos/videos 'Not in any album', so adding a Multi-select dropdown with all albums listed inside this 'DISPLAY OPTIONS' groups could probably be a nice possibility.

**Comment by mannok** [comment]
Apart from filtering   I think ordering option is also useful. Currently it is lacking from shared album

**Comment by tobiasdiez** [comment]
Coming from https://github.com/immich-app/immich/discussions/19273, another aspect is a negated filter "not in album xyz" to find pictures in say the generic "Camera" album that still have to be processed.

**Reply by scaredmushroom** [reply]
I second this. Generally it would be cool to be able to say: is in album x and y, but not in album z. This would help me a lot!

**Reply by teto** [reply]
exactly my usecase. Seems like one has to create a "prxoy" album to enable automatic upload. Once uploaded in that album, I haven't found a way to see if a picture in the "proxy" album already belongs to other albums (processed) or not.
As a general UX thing, I would to see if pictures belong to one or several albums, possibly which ones, at a glance.

**Comment by cswrd** [comment]
I think a filter functionality provided by a funnel like symbol would be helpful on many pages and could probably be very generic from a look and feel. In fact it might probably be the same as the modal view in the extended global search function, but apply to the current view only. It could be helpful in the albums list, individual albums themself, the map and maybe more.

The existing quick filters in the albums list to show e.g. only own albums should be looked at together with what I said above and could be extended also. Both in terms of providing it on more views as well as what you can quickly filter. For example a ♥ button would be quite helpful there, e.g. to show only favorite albums or favorite assets in an album quickly.

**Reply by hugo-id** [reply]
Having advanced _text_ search as detailed elsewhere would be great, but for maximum _adoption_ nothing will beat having proper, GUI filtering _everywhere_

<img width="1652" height="342" alt="image" src="https://github.com/user-attachments/assets/43ef50ac-562a-4be2-818d-626d88dc50de" />

I have only spun up immich as of 4 days ago but already this is an annoyance:
- Show only photos taken by ____ in shared album
- Show only photos of ______ in my favourites
- Hide photos with _____ in it in main view
- Show photos with _____ tag taken in FY 2024 (ie "receipts", or "photo scans")

Filtering is the kind of GUI thing that will make my mum use Immich

**Comment by nrother** [comment]
This seems to be a UI problem only, manually crafting a URL á la `/search?query={%22albumIds%22%3A[%228a56e14f-4255-4531-b3b1-bd066616f8bf%22]}` works fine. Surprisingly, filtering by album is also mentioned in the [docs](https://immich.app/docs/features/searching/):

> These filters include location, one or more faces, specific albums, and more.

I suggest adding a "filter by album" function, very alike the the "filter by tag" UI that is currently there.

**Reply by aljora** [reply]
Web portion seems to be in review: #22788

**Comment by scaredmushroom** [comment]
Would it be feasible to provide an advanced search syntax? Something like: "album:Camera and not album:Vacation and face:Johanna". Currently sorting bigger amounts of pictures is not really possible for me and such an addition would enable so many options.

**Reply by WirtsLegs** [reply]
This would be the ideal way to go I think

I have so many assets that ideally should end up in multiple albums but trying to find them without just trying to re-add all the appropriate images is a pain currently

Ability to search for items not in a specific album, or in album x but not y would be a major convenience factor for me

**Reply by SinTan1729** [reply]
This will be great. It would make finding stuff so much easier for me. PhotoPrism has this to some degree, but Immich should definitely offer an advanced search syntax.

**Comment by Idan37S** [comment]
This feature seems so simple and so useful,
I really hope it will be implemented,
I think it would help a lot of people.

Not 100% related to this feature, but somewhat related,
It would also help if there was a row of relevant people inside an album page, and once you click on of them it filters automatically the images in that album by that person,
It would really boost the user experience!

**Comment by janeoa** [comment]
Hello immich team! Is this feature on its way and is somehow blocked? If it is a matter of not having a person assigned, I can try doing the feature. How do I get assigned?

**Comment by scr4bble** [comment]
I'd also like to highlight the UX when starting from an already opened album.

#25740 was closed because media-type filtering is available in Search, but that doesn't solve this use case: when viewing an album, there is currently no way to filter its contents, e.g. show only videos.

The suggested Album filter / multi-select in Search would solve part of this. It would be even better if the same filter UI could be opened directly from an album, with the current album already selected.

So the flow could simply be:

Album → Filter → Media type: Video

It doesn't necessarily need to filter inside the album view — opening Search with the current album preselected would work just as well.

**Comment by bataq** [comment]
Status update on this request: the web side looks done, the mobile side is not.

On web, filtering a search by album works now — it went in via #21001, which was closed as completed. It also combines with other filters.
The mobile app's search filters still have no album option, so there is no way to narrow down a large album from the app.

So this request is effectively mobile-only at this point.
